src/client.ts

```diff
@@ -56,42 +56,47 @@ export class Client {
    * ```
    */
   constructor(clientConfiguration?: Partial<ClientConfiguration>) {
-    this.clientConfiguration = {
+    this.#clientConfiguration = {
       ...defaultClientConfiguration,
       ...clientConfiguration,
       secret: this.#getSecret(clientConfiguration),
     };
+
     // ensure the network timeout > ClientConfiguration.queryTimeoutMillis so we don't
     // terminate connections on active queries.
-    const timeout = this.clientConfiguration.timeout_ms + 10_000;
+    const timeout = this.#clientConfiguration.timeout_ms + 10_000;
     const agentSettings = {
-      maxSockets: this.clientConfiguration.max_conns,
-      maxFreeSockets: this.clientConfiguration.max_conns,
+      maxSockets: this.#clientConfiguration.max_conns,
+      maxFreeSockets: this.#clientConfiguration.max_conns,
       timeout,
       // release socket for usage after 4s of inactivity. Must be less than Fauna's server
       // side idle timeout of 5 seconds.
       freeSocketTimeout: 4000,
       keepAlive: true,
     };
     this.client = axios.create({
-      baseURL: this.clientConfiguration.endpoint.toString(),
+      baseURL: this.#clientConfiguration.endpoint.toString(),
       timeout,
     });
     this.client.defaults.httpAgent = new Agent(agentSettings);
     this.client.defaults.httpsAgent = new HttpsAgent(agentSettings);
-    this.client.defaults.headers.common[
-      "Authorization"
-    ] = `Bearer ${this.clientConfiguration.secret}`;
     this.client.defaults.headers.common["Content-Type"] = "application/json";
     // WIP - presently core will default to tagged; hardcode to simple for now
     // until we get back to work on the JS driver.
     this.client.defaults.headers.common["X-Format"] = "simple";
     this.#setHeaders(
-      this.clientConfiguration,
+      this.#clientConfiguration,
       this.client.defaults.headers.common
     );
   }
 
+  /**
+   * Return the current {@link endpoint} for the Fauna Client
+   */
+  public get endpoint(): URL {
+    return this.#clientConfiguration.endpoint;
+  }
```

Comment:
Let's just let them get the client configuration back.

```suggestion
  /**
   * Return the {@link ClientConfiguration} of this client, save for the secret.
   */
  public get clientConfiguration(): Omit<ClientConfiguration, "secret"> {
    const { secret, ...rest } = this.#clientConfiguration;
    return rest;
  }
```

Reply:
See utility types: https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys